app/models/transaction.rb

```diff
@@ -55,7 +56,8 @@ def total
   validates :category, presence: true, unless: :residue?
   validates :bank_account, presence: true
   validates :transaction_type, inclusion: { in: TRANSACTION_TYPES, allow_blank: true }
-  validates :created_at, presence: true, on: :update
+  validates :date, presence: true, on: :update
```

Comment:
@yunixon date should be filled on creation too. Just fill it with current date if it is blank on `before_validate` callback